:pencil: (proofread) Revise mentoring docs (#273)
exercism/docs: mentoring/how_to_give_great_feedback.md

You're about to give your first feedback to a student.
This can be exciting but also pretty daunting.
A lot of (the best!) mentors suffer from [Impostor Syndrome](https://en.wikipedia.org/wiki/Impostor_syndrome) at this stage and give up before they start.
If you feel like that, it's totally understandable but also almost always unnecessary.
By the fact that you're worrying about whether your feedback is going to be good enough, it's likely that you'll give thoughtful and humble comments that students will really appreciate.

> And in the worst case, a student can always end the discussion and wait for another mentor. So let's give it a go!

Reviewer: ```suggestion
And in the worst case, a student can always end the discussion and send it back to the queue for another mentor. So give it a go!
```

## Mindset

The first thing to keep in mind is having the right mindset as a mentor.

> Students are hoping you can help improve their knowledge and tell them things that they don't know.

Reviewer: ```suggestion
Students are hoping you can help improve their knowledge and tell them things that they do not know.
```

Your job isn't to "mark" their work, it's to use their solution as a basis for unlocking ideas that they are not familiar with or are struggling with.

It's also worth remembering that most Exercism users are experienced developers learning a new language.

> So although their code might look bad to you, that may well be because they simply aren't experienced writing _this_ language.

Reviewer: ```suggestion
So although their code might look _bad_ to you, that may well be because they simply are not experienced writing _this_ language.
```

Similar code in a different language might be totally idiomatic and correct.

> Conversations tend to be more productive when they are approached as two peers discussing a topic rather than one person showing off their knowledge to another.

Reviewer: I like the comma that was here before.  The sentence could have ended where that comma was, and so the pause is to extend on the idea that was presented up to that point.

Reviewer 2: I agree with restoring the comma.

Author: Reverted.

## Addressing the student's request vs teaching a new idea

When a student submits a solution for mentoring, they are prompted to tell the mentor what they are hoping to get from it.

> It's really important that you as the mentor read this and try and address it in your feedback.

Reviewer: ```suggestion
It is really important that you, as the mentor, read this and try and address it in your feedback.
```

Different mentors have different approaches to whether they give "hints" or "explanations".
As a general rule, aim for somewhere between the two.

> Ensure that you tell the student enough that they can understand what they should try but try not to just give them the answer and tell them why it works.

Reviewer 2: I prefer a comma here.

```suggestion
Ensure that you tell the student enough that they can understand what they should try, but try not to just give them the answer and tell them why it works.
```

b = a ? 1 : 2
```

> You could simply say, "Write `b = a ? 1 : 2` instead of the `if` statement on L10" but that wouldn't be helping a student understand the _why_ or have a lightbulb moment of their own.

Reviewer 2: I prefer a comma here.

```suggestion
You could simply say, "Write `b = a ? 1 : 2` instead of the `if` statement on L10," but that wouldn't be helping a student understand the _why_ or have a lightbulb moment of their own.
```

Reviewer: I like the two sentences.  We could.  But that would not be helping…

> Instead it would be much better to introduce the idea of a ternary and pose the student a challenge.

Reviewer: ```suggestion
Instead it would be much better to introduce the idea of a ternary, posing a challenge to the student.
```

Author: This suggested change would transform the final clause into the passive voice which most English-language styles guides discourage. I don't have a linguistic style guide for the site I can reference. Perhaps we could get a confirming opinion to accept this change @ErikSchierboom ?

Reviewer 2: This is a link to our style guide: https://exercism.org/docs/building/markdown/style-guide

Author: But your opinion on this particular change (not discussed in the style guide)?

Reviewer 2: I'm fine with either. Your call.

Reviewer: The way that I took the sentence here is that we would introduce the idea of a ternary and then pose a challenge to the student, which would be another action after the introduction.  I think introducing the ternary would pose a challenge to the student in the same action, so the comma and change of wording may communicate this.  Not passive, the action is proposed, the effect presented.

Author: I disagree in two regards.

Insinuating that the act of introducing *is* posing a challenge is directly contrary to the directions immediately preceding and example following:

> You could simply say, "Write `b = a ? 1 : 2` instead of the `if` statement on L10," but that wouldn't be helping a student understand the _why_ or have a lightbulb moment of their own.

> For example: "I think you could reduce the Lines 10-14 down to one line using the Ternary Operator. How might that look?"

Secondly, the end clause is passive. An active version would be some version of "challenging the student" (not the inversion you have presented).

For example: "I think you could reduce the Lines 10-14 down to one line using the Ternary Operator. How might that look?"
A student will then have to play with that and solve it themselves, but you've given them enough info to work with.

On the other end of the scale, saying "Reduce lines 10-14 to one line" wouldn't give a student any indication of how that's achievable, and so could be incredibly frustrating for them.
Try to ensure you give enough information for a student to progress.

## How much feedback should you give

> Different mentors have different styles but our default suggestion is to give the most important one to three pieces of feedback on any iteration.

Reviewer 2: I prefer the comma here.

```suggestion
Different mentors have different styles, but our default suggestion is to give the most important one to three pieces of feedback on any iteration.
```

> If the student submits a new iteration with those solved, you could give another one to three pieces of feedback.

Reviewer 2: ```suggestion
If the student submits a new iteration with those solved, you could then give another one to three pieces of feedback.
```
